pybind11/__main__.py

```diff
@@ -2,12 +2,35 @@
 from __future__ import annotations
 
 import argparse
+import re
 import sys
 import sysconfig
 
 from ._version import __version__
 from .commands import get_cmake_dir, get_include, get_pkgconfig_dir
 
+# This is the conditional used for os.path being posixpath
+if "posix" in sys.builtin_module_names:
+    from shlex import quote
+elif "nt" in sys.builtin_module_names:
+    # See https://github.com/mesonbuild/meson/blob/db22551ed9d2dd7889abea01cc1c7bba02bf1c75/mesonbuild/utils/universal.py#L1092-L1121
+    # and the original documents:
+    # https://docs.microsoft.com/en-us/cpp/c-language/parsing-c-command-line-arguments and
+    # https://blogs.msdn.microsoft.com/twistylittlepassagesallalike/2011/04/23/everyone-quotes-command-line-arguments-the-wrong-way/
+    UNSAFE = re.compile("[ \t\n\r]")
+
+    def quote(s: str) -> str:
+        if s and not UNSAFE.search(s):
+            return s
+
+        # Paths cannot contain a '"' on Windows, so we don't need to worry
+        # about nuanced counting here.
+        return f'"{s}"'
```

Comment:
Does this handle a trailing backslash correctly? I see all the internally-generated paths use `os.path.join`, and are probably safe, but I'm not 100% sure about `sysconfig.get_path`.

Reply:
It doesn't. We could use `if s.endswith("\\") s += "\\"` in line 28 to fix this. 
Cmd does not accept escaped `\\` in general, but in front of a terminal `"` it's ok.

Reply:
```suggestion
        # Paths cannot contain a '"' on Windows, so we don't need to worry
        # about nuanced counting here.
        return f'"{s}\\"' if s.endswith("\\") else f'"{s}"'
```
?

Reply:
Unless you want to support something like `c:\` as an include path, maybe stripping trailing backslashes would be safe enough?